eldar/query.py

```diff
@@ -22,6 +23,23 @@ def preprocess(self, doc):
             doc = doc.lower()
         if self.ignore_accent:
             doc = unidecode(doc)
+
+        # demojise query
+        doc = emoji.demojize(doc, language='en', delimiters=("::__", "__::"))
+
+        emoji_matches = re.findall(r"::__[a-zA-Z0-9._-]+__::", doc)
+
+        for match in emoji_matches:
+            try:
+                # convert emoji to original format to compute the actual length of it
+                match_as_emoji = emoji.emojize(match.replace("::__", ":").replace("__::", ":"))
+
+                # replace emoji with whitespace equal to the length of emoji
+                doc = re.sub(match, " "*len(match_as_emoji), doc)
+
+            except Exception:
+                doc = re.sub(match, " ", doc)
```

Comment:
We actually don't need al this as we are just looking to convert the emoji into a format we can run search over e.g.

This is a test 😊 -> This is a test :smiling_face_with_smiling_eyes:

Then if we convert the search query using a similar method, it will work for searching. To do this you will need to modify the `parse_query` function to also demojise the query (e.g. `query = emoji.demojize(query, language='en'))